### PR TITLE
Move `monitoring` to specific components

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR moves `monitoring` of the specific components from charts to golang structs.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/2754

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/monitoring.go [hunk 1]
```diff
@@ -84,15 +84,25 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		b.Shoot.Components.ControlPlane.ResourceManager,
 	}
 
+	if b.Shoot.IsShootControlPlaneLoggingEnabled(b.Config) && gardenlethelper.IsValiEnabled(b.Config) {
```
Comment (shafeeqes) [inline pkg/operation/botanist/monitoring.go:87]: Why was `b.Shoot.IsShootControlPlaneLoggingEnabled(b.Config)` added?

**Review by shafeeqes — COMMENTED** [review]
Thanks for the PR, only one question.

File: pkg/operation/botanist/monitoring.go (hunk 1 above)
Comment (acumino) [reply pkg/operation/botanist/monitoring.go:87]: `IsShootControlPlaneLoggingEnabled ` check if logging is enabled in gardenlet config, if it is not there is no need of vali.

**Review by shafeeqes — COMMENTED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

**Pull request merged**